CMSgov/mdct-mcr
# finish dashboard functionality

## Description

Dashboard needed to be wired up to the API. 

Changes:
- Add `FormViewProvider` to pass formJson dynamically to report form pages
- Clean up and finalize `Dashboard` structure and role-based UI variances
- Modify `Dashboard` to accept non-state users and render dashboard view based on local storage value.
- Add `AddEditProgramModal` and wire up to API
- Add `DeleteProgramModal` and wire up to API
- Tweak underlying `Modal` code

_Note: There are also a few other fast-follow tickets for which I will add new tickets for the team to pick up._

### How to test

1. Sign in as state user and navigate to dashboard. You can create and edit reports now. Hydration is not hooked up, but everything else works. You cannot delete programs.
2. Sign in as admin user and navigate to dashboard. You should see nothing. Go to browser local storage and set a new item with key = `selectedState` and value = `the 2-letter code of whatever state for which you have a report in the local db`. You should not be able to edit this report or create any new reports. You should be able to delete a report.

### Changed Dependencies

n/a

## Code author checklist

## Reviewer checklist (two different people)
